x-pack/legacy/plugins/apm/server/new-platform/plugin.ts

```diff
@@ -7,12 +7,13 @@
 import { InternalCoreSetup } from 'src/core/server';
 import { makeApmUsageCollector } from '../lib/apm_telemetry';
 import { CoreSetupWithUsageCollector } from '../lib/apm_telemetry/make_apm_usage_collector';
+import { createApmAgentConfigurationIndex } from '../lib/settings/agent_configuration/create_agent_config_index';
 import { createApmApi } from '../routes/create_apm_api';
 
 export class Plugin {
   public setup(core: InternalCoreSetup) {
     createApmApi().init(core);
-
+    createApmAgentConfigurationIndex(core);
```

Comment:
This issue made it clear to me that index creation belongs in this setup step.

Reply:
I seem to recall we did that initially and then we were asked not to do it during startup. Or was that only for the index pattern?

Reply:
If this is okay for platform I'm +1. Seems like the right place to do it 👍

Reply:
it was the index pattern before because it prevented the welcome screen, but this index doesn't have the same effect.